internal/pkg/agent/application/upgrade/step_unpack.go

```diff
@@ -219,19 +219,26 @@ func getPackageMetadataFromZip(archivePath string) (packageMetadata, error) {
 }
 
 func skipFnFromZip(log *logger.Logger, r *zip.ReadCloser, detectedFlavor string, fileNamePrefix string, versionedHome string, registry map[string][]string) (install.SkipFn, error) {
+	acceptAllFn := func(relPath string) bool { return false }
 	if detectedFlavor == "" {
 		// no flavor don't skip anything
-		return func(relPath string) bool { return false }, nil
+		return acceptAllFn, nil
 	}
 
 	flavor, err := install.Flavor(detectedFlavor, "", registry)
 	if err != nil {
 		if errors.Is(err, install.ErrUnknownFlavor) {
 			// unknown flavor fallback to copy all
-			return func(relPath string) bool { return false }, nil
+			return acceptAllFn, nil
 		}
 		return nil, err
```

Comment:
since I was part of the initial investigation of why these tests were failing, something that was clearly missing from this code, is error wrapping with distinguishable messages. Could you please introduce proper error wrapping in the returned errors? 🙂

Reply:
This is an important bit that would allow us to have a better error than "open : The system cannot find the file specified."